### PR TITLE
Netlify plugins

### PR DESCRIPTION
**Pull request by kewitham**
This PR adds support for the following HTML partials so they render on Netlify:
### Accordion
**Before**
image

**After**
image

### Asset
**Before**
image

**After**
image
### Details
**Before**
image
**After**
image

### Footnote
**Before**
image

**After**
image

### Figure
**Before**
image

**After**
image

### Icon list
**Before**
image

**After**
image

### CODE REVIEW

File: _assets/js/netlify/makeTemplate.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import formatPublishDates from './formatPublishDates';
 
 /** @jsx h */
 
-const makeTemplate = (data) => {
+function makeTemplate(data) {
```
Comment (okkays) [inline _assets/js/netlify/makeTemplate.js:6]: 🔥

File: _assets/js/netlify/markdownWidgets/collapsible.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        .on('template', (tpl) => this.tpls.push(tpl))
+        // note that we cannot use arrow function because we need `this`
+        .on('tag:endcollapsible', function () {
+          this.stop();
+        })
+        .on('end', () => {
+          throw new Error(`tag ${tagToken.getText()} not closed`);
+        })
+        .start();
+    },
```
Comment (okkays) [inline _assets/js/netlify/markdownWidgets/collapsible.js:17]: this function is the same across multiple files.

Any way to abstract it?

File: _assets/js/netlify/markdownWidgets/listItem.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        })
+        .on('end', () => {
+          throw new Error(`tag ${tagToken.getText()} not closed`);
+        })
+        .start();
+    },
+    *render(context, emitter) {
+      const iconType = context._iconType;
+      emitter.write(
+        `<li class='usa-icon-list__item'><div class='usa-icon-list__icon'><div class='usa-icon' aria-hidden='true' style="background-image:url('${window.location.origin}/assets/img/usa-icons/${iconType}.svg')"></div></div><div class='usa-icon-list__content'>`
```
Comment (okkays) [inline _assets/js/netlify/markdownWidgets/listItem.js:20]: this doesn't have to be on one line (here and above). With template literals (\`), I think you can do:

```suggestion
        `<li class='usa-icon-list__item'>
          <div class='usa-icon-list__icon'>
            <div class='usa-icon' aria-hidden='true' 
                style="background-image:url('${window.location.origin}/assets/img/usa-icons/${iconType}.svg')"
                >
            </div>
          </div>
          <div class='usa-icon-list__content'>
        `
```

**Review by okkays — CHANGES_REQUESTED** [review]
Nice work!

A bit of confusion on the PR title - this is for HTML partials (includes under `_includes/*.html`) or a selection of widgets (ruby plugins under `_plugins/*.rb`)?

**Comment by okkays** [comment]
general note: strange that the pages build is still running, as this was pushed last night:
image

File: _assets/js/netlify/markdownWidgets/collapsible.js (hunk 2 above)
Comment (kewitham) [reply _assets/js/netlify/markdownWidgets/collapsible.js]: updated!

File: _assets/images/30X48ClrFlrSpc.jpg
[the remark this answers is not in the document]
Comment (kewitham) [reply _assets/images/30X48ClrFlrSpc.jpg:1]: updated to remove image changes

**Comment by okkays** [comment]
Just for @vwilliamsn and @c2nelson's context, I've asked @kewitham to remove the image changes for now until we can find a way to render them without duplicating them in the repo.

So the previews should render like in the screenshots, but the images might be broken until we find a proper fix.

**Review by okkays — APPROVED** [review]

**Pull request merged**